src/solvers/flattening/boolbv_index.cpp

```diff
@@ -256,7 +256,7 @@ bvt boolbvt::convert_index(const index_exprt &expr)
 
       typet constant_type=index.type(); // type of index operand
 
-      assert(array_size>0);
```

Comment:
My only request is perhaps a better error message here? This shows what's going on, but not why. But I'm not too fussed about it to be honest.